staking/programs/integrity-pool/src/state/pool.rs

```diff
@@ -179,19 +184,29 @@ impl PoolData {
             IntegrityPoolError::OutdatedPublisherCaps
         );
 
-        for publisher_cap in publisher_caps.caps.iter() {
-            if self.get_publisher_index(&publisher_cap.pubkey).is_err() {
-                self.add_publisher(&publisher_cap.pubkey)?;
-            }
+        for epoch in self.last_updated_epoch..current_epoch {
+            let event =
+                self.get_event_mut((self.num_events + epoch - self.last_updated_epoch) as usize);
+            event.epoch = epoch;
+            event.y = y;
         }
 
-        // create the reward event for last_updated_epoch using current del_state before updating
-        // which corresponds to del_state at the last_updated_epoch
-        self.create_reward_event(publisher_caps, y, self.last_updated_epoch)?;
-
         let epochs_passed = current_epoch - self.last_updated_epoch;
+        let mut i = 0;
+
+        while i < MAX_PUBLISHERS && self.publishers[i] != Pubkey::default() {
+            let cap_index = Self::get_publisher_cap_index(&self.publishers[i], publisher_caps)?;
```

Comment:
I think this implies that a publisher that has been added is always in the publisher_caps which is probably not always true.

Reply:
Fixed.